C/plugins/storage/postgres/connection.cpp

```diff
@@ -3312,3 +3312,422 @@ const char *p;
+                        {
+                                raiseError("retrieve", "Failed to parse JSON payload");
+                                return -1;
+                        }
+			if (!document.HasMember("schema"))
+                        {
+				Logger::getLogger()->error("schema absent from payload, so exiting ");
+				return -1;
+			}
+			else
+			{
+				schema = document["schema"].GetString();
+				if (!document.HasMember("service"))
+				{
+					Logger::getLogger()->debug("service absent from payload, so exiting");
+                                        return -1;
+				}
+				std::string service = document["service"].GetString();	
+				if (service.empty())
+				{
+					Logger::getLogger()->debug("empty service name, so exiting");
+                                        return -1;
+				}
+
+				if (!document.HasMember("version"))
+                        	{
+					Logger::getLogger()->debug("version absent from payload, so exiting");
+                                	return -1;
+                        	}
+				else
+				{
+					version = document["version"].GetInt();
+
+					// insert payload in 
+			//std::string s = "insert into fledge.service_schema(name, version, definition) values ('" + schema + "', " + to_string(version) + ", " + "'" + payload + "') ;" ;
+
+			//		rowsAffectedLastCommand = purgeOperation(s.c_str(), logSection, "insert in fledge.service_schema  ", false);
+
+					if (!schema.empty())
+					{
+						std::string results;
+						std::string name = service + "_" + schema;
+						if (findSchemaFromDB(name, results))
+						{
+							if (parseDatabaseStorageSchema(version, results, columnMap, indexMap))
+							{
+								//comparePayloads(payload, results);
+								int i;
+
+							}
+						}
+							
+						std::string queryToCreateSchema = "create schema if not exists " + schema + ";" ;
+	                                        rowsAffectedLastCommand = purgeOperation(queryToCreateSchema.c_str(), logSection, "Create Schema if not exists ", false);
+					}
+
+				}
+				if (!document.HasMember("tables"))
+                        	{
+					Logger::getLogger()->error("tables section absent from payload, so exiting");
+                                	return -1;
+                        	}
+				else
+				{
+					Value& tables = document["tables"];
+					if (!tables.IsArray())
+                                	{
+                                        	raiseError("retrieve", "The property return must be an array");
+                                        	return -1;
+                                	}
+					else
+                          		{
+                                		for (rapidjson::SizeType i = 0; i < tables.Size(); i++)
+						{
+							std::string name = tables[i]["name"].GetString();
+							Value& columns = tables[i]["columns"];
+							vector<std::string> index;
+							vector<columnRec> vecColumns;
+
+							std::string sql, sqlIdx;
```

Comment:
The SQLBuffer class is designed for this and can make creating the SQL commands easier and more efficient than using the string class.

Reply:
purgeOperation makes use of SQLBuffer class only to execute the query , argument it requires is std::string